server/src/main/java/org/jetbrains/bsp/bazel/server/sync/TargetKindResolver.java

```diff
@@ -7,22 +7,29 @@
 import org.jetbrains.bsp.bazel.server.sync.model.Tag;
 
 public class TargetKindResolver {
-  private final Map<String, Tag> ruleSuffixToTargetType =
+  private static final Set<Tag> LIBRARY = HashSet.of(Tag.LIBRARY);
+  private static final Map<String, Set<Tag>> ruleSuffixToTargetType =
       Map.of(
-          "library", Tag.LIBRARY,
-          "binary", Tag.APPLICATION,
-          "test", Tag.TEST);
+          "library", LIBRARY,
+          "binary", HashSet.of(Tag.APPLICATION),
+          "test", HashSet.of(Tag.TEST));
+
+  private static final Set<Tag> NO_IDE = HashSet.of(Tag.NO_IDE);
 
   public Set<Tag> resolveTags(TargetInfo targetInfo) {
+    if (targetInfo.getKind().equals("resources_union")) {
+      return LIBRARY;
```

Comment:
no this actually needs to be a "normal" target.  I'm pretty sure this is working around a twitter specific bazel quirk, but this is needed to maintain the correct dependency graph, otherwise things that depend on resources won't correctly import.

Reply:
i meant that this rule should support the `no-ide` tag as well as any other, but if there is no such use case then fine